### PR TITLE
Feature/hw 53100 core commit transfer

### PR DESCRIPTION
**Pull request by oleksii-lubianyi-epam**

### CODE REVIEW

File: Sources/Hyperwallet.swift [hunk 1]
```diff
@@ -248,6 +248,31 @@ public final class Hyperwallet {
...
+    ///
+    /// This function will request a new authentication token via `HyperwalletAuthenticationTokenProvider`
+    /// if the current one is expired or is about to expire.
+    ///
+    /// - Parameters:
+    ///   - transferMethodToken: the Hyperwallet specific unique identifier for the `HyperwalletTransfer`
+    ///                          being commited
+    ///   - notes: a note regarding the status change
+    ///   - completion: the callback handler of responses from the Hyperwallet platform
+    public func commitTransfer(transferMethodToken: String,
```
Comment (syyang-hw) [inline Sources/Hyperwallet.swift]: transferToken

File: Sources/Hyperwallet.swift [hunk 2]
```diff
@@ -248,6 +248,31 @@ public final class Hyperwallet {
...
+    ///
+    /// The `completion: @escaping (HyperwalletStatusTransition?, HyperwalletErrorType?) -> Void` that is passed in to
+    /// this method invocation will receive the successful response(HyperwalletStatusTransition) or
+    /// error(HyperwalletErrorType) from processing the request.
+    ///
+    /// This function will request a new authentication token via `HyperwalletAuthenticationTokenProvider`
+    /// if the current one is expired or is about to expire.
+    ///
+    /// - Parameters:
+    ///   - transferMethodToken: the Hyperwallet specific unique identifier for the `HyperwalletTransfer`
```
Comment (syyang-hw) [inline Sources/Hyperwallet.swift]: transferToken

File: Sources/Hyperwallet.swift [hunk 3]
```diff
@@ -248,6 +248,31 @@ public final class Hyperwallet {
...
+    ///                          being commited
+    ///   - notes: a note regarding the status change
+    ///   - completion: the callback handler of responses from the Hyperwallet platform
+    public func commitTransfer(transferMethodToken: String,
+                               notes: String? = nil,
+                               completion: @escaping (HyperwalletStatusTransition?, HyperwalletErrorType?) -> Void) {
+        let statusTransition = HyperwalletStatusTransition(transition: .scheduled)
+        statusTransition.notes = notes
+        httpTransaction.performRest(httpMethod: .post,
+                                    urlPath: "transfers/\(transferMethodToken)/status-transitions",
```
Comment (syyang-hw) [inline Sources/Hyperwallet.swift]: transferToken

File: Sources/Hyperwallet.swift (hunk 1 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Hyperwallet.swift]: Renamed

File: Sources/Hyperwallet.swift (hunk 2 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Hyperwallet.swift]: Renamed

File: Sources/Hyperwallet.swift (hunk 3 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Hyperwallet.swift]: Renamed

File: Sources/Hyperwallet.swift [hunk 4]
```diff
@@ -248,6 +248,31 @@ public final class Hyperwallet {
...
+    ///
+    /// This function will request a new authentication token via `HyperwalletAuthenticationTokenProvider`
+    /// if the current one is expired or is about to expire.
+    ///
+    /// - Parameters:
+    ///   - transferToken: the Hyperwallet specific unique identifier for the `HyperwalletTransfer`
+    ///                    being commited
+    ///   - notes: a note regarding the status change
+    ///   - completion: the callback handler of responses from the Hyperwallet platform
+    public func commitTransfer(transferToken: String,
```
Comment (jsaini-hw) [inline Sources/Hyperwallet.swift]: rename it to scheduleTransfer to keep it consistent with the UI

File: Sources/Hyperwallet.swift [hunk 5]
```diff
@@ -248,6 +248,31 @@ public final class Hyperwallet {
                                     completionHandler: completion)
     }
 
+    /// Commits the `HyperwalletTransfer` linked to the transfer method token specified.
```
Comment (jsaini-hw) [inline Sources/Hyperwallet.swift]: Rename commits to Schedule

File: Tests/Responses/Transfer/CommitTransferResponse.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (jsaini-hw) [inline Tests/Responses/Transfer/CommitTransferResponse.json]: rename file to ScheduleTransferResponse

File: Tests/Transfer/HyperwalletTransfer.swift [hunk 7]
```diff
@@ -114,6 +114,39 @@ class HyperwalletTransferTests: XCTestCase {
         verifyTransferResponse(transferResponse)
         XCTAssertEqual(transferResponse?.token, "trf-123456")
     }
+
+    func testCommitTransfer_success() {
```
Comment (jsaini-hw) [inline Tests/Transfer/HyperwalletTransfer.swift]: rename test to testScheduleTransfer_success

File: Tests/Transfer/HyperwalletTransfer.swift [hunk 8]
```diff
@@ -114,6 +114,39 @@ class HyperwalletTransferTests: XCTestCase {
...
+        //When
+        Hyperwallet.shared.commitTransfer(transferToken: "trf-123456", completion: { (result, error) in
+            statusTransitionResponse = result
+            errorResponse = error
+            expectation.fulfill()
+        })
+        wait(for: [expectation], timeout: 1)
+
+        //Then
+        guard errorResponse == nil else {
```
Comment (jsaini-hw) [inline Tests/Transfer/HyperwalletTransfer.swift]: Use ```  XCTAssertNil(errorResponse, "The `errorResponse` should be nil") ```

File: Tests/Transfer/HyperwalletTransfer.swift [hunk 9]
```diff
@@ -114,6 +114,39 @@ class HyperwalletTransferTests: XCTestCase {
...
+            XCTAssertTrue(false, "The `errorResponse` should be nil")
+            return
+        }
+
+        guard let statusResponse = statusTransitionResponse else {
```
Comment (jsaini-hw) [inline Tests/Transfer/HyperwalletTransfer.swift]: Use ``` XCTAssertNotNil(statusTransitionResponse, "The `statusResponse` should not be nil") ```

File: Sources/Hyperwallet.swift (hunk 4 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Hyperwallet.swift]: Renamed

File: Sources/Hyperwallet.swift (hunk 5 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Hyperwallet.swift]: Renamed

File: Tests/Responses/Transfer/CommitTransferResponse.json (hunk 6 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Responses/Transfer/CommitTransferResponse.json]: Renamed

File: Tests/Transfer/HyperwalletTransfer.swift (hunk 7 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Transfer/HyperwalletTransfer.swift]: Renamed

File: Tests/Transfer/HyperwalletTransfer.swift (hunk 8 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Transfer/HyperwalletTransfer.swift]: Updated

File: Tests/Transfer/HyperwalletTransfer.swift (hunk 9 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Transfer/HyperwalletTransfer.swift]: Updated

**Comment by oleksii-lubianyi-epam** [comment]
Created branch with proper name. New PR to be created

**Pull request closed without merging**